presto-kudu/src/main/java/io/prestosql/plugin/kudu/KuduMetadata.java

```diff
@@ -375,7 +418,48 @@ public boolean usesLegacyTableLayouts()
     public ConnectorTableProperties getTableProperties(ConnectorSession session, ConnectorTableHandle table)
     {
         KuduTableHandle handle = (KuduTableHandle) table;
-        return new ConnectorTableProperties(handle.getConstraint(), Optional.empty(), Optional.empty(), Optional.empty(), ImmutableList.of());
+        KuduTable kuduTable = handle.getTable(clientSession);
+
+        Optional<ConnectorTablePartitioning> tablePartitioning = Optional.empty();
+        Optional<Set<ColumnHandle>> partitioningColumns = Optional.empty();
+        List<LocalProperty<ColumnHandle>> localProperties = ImmutableList.of();
+
+        if (isKuduGroupedExecutionEnabled(session)
+                && isTableSupportGroupedExecution(kuduTable)
+                && handle.getBucketCount().isPresent()) {
```

Comment:
Isn't it redundant ?

Reply:
You are right, bucketCount will not be empty once `PartitionSchema.getHashBucketSchemas` is present.